packages/backend/src/nest/storage/base.store.ts

```diff
@@ -0,0 +1,27 @@
+import KeyValueStore from 'orbit-db-kvstore'
+import Store from 'orbit-db-store'
+import EventStore from 'orbit-db-eventstore'
+import { EventEmitter } from 'events'
+
+export default abstract class StoreBase<V, S extends KeyValueStore<V> | EventStore<V>> extends EventEmitter {
+  protected abstract store: S | undefined
+
+  getStore() {
+    if (!this.store) {
+      throw new Error('Store not initialized')
+    }
+    return this.store
+  }
+
+  getAddress(): Store['address'] {
+    return this.getStore().address
+  }
+
+  async close(): Promise<void> {
+    await this.store?.close()
+  }
+
+  abstract init(): Promise<void>
+  abstract addEntry(entry: V): Promise<V | undefined>
```

Comment:
To me, this abstraction doesn't seem like the best fit. KeyValueStore has a map-like interface (e.g. `setEntry` `getEntry`), while EventStore has a list-like interface (e.g. `addEntry`, `getEntries`). So perhaps an additional base class for each type of store would be more appropriate or just remove `addEntry` from this base class which is probably what I'd recommend for now.

Reply:
Agreed.  We could have two layers of base class:

1. A base store that has the universal stuff like close, init, etc.
2. Individual base classes for each type of store that more closely follows the underlying orbitdb store

Reply:
I was going for more abstract/general interface, hence the 'addEntry'. It would implement 'setting' for kvstore and 'adding' for event store. However you may have a point so I'll change that.

Reply:
Btw. do you have a strong preference about `setEntry` params for the base store class of kvstore? If we want it to be closer to orbitdb store method (and more generic) then we should pass key and value here. Currently our stores use key taken from value (e.g meta.id/userProfile.pubKey) so technically we can pass only value.
I'd lean towards key+value.

Reply:
I was thinking about that too. I think probably key+value is easier to read and understand what's going on